Zotero/Scenes/Master/Collections/ViewModels/CollectionsActionHandler.swift

```diff
@@ -327,6 +361,36 @@ final class CollectionsActionHandler: ViewModelActionHandler, BackgroundDbProces
                     break
                 }
             })
+
+            func refreshCollectionItemCounts(for collectionKeys: Set<String>, in viewModel: ViewModel<CollectionsActionHandler>, handler: CollectionsActionHandler) {
+                guard !collectionKeys.isEmpty else { return }
+                var counts: [String: Int] = [:]
+                for key in collectionKeys {
+                    let request = ReadItemsDbRequest(collectionId: .collection(key), libraryId: viewModel.state.library.identifier)
+                    do {
+                        let results = try handler.dbStorage.perform(request: request, on: .main)
+                        counts[key] = results.freeze().count
```

Comment:
Shouldn't need to freeze if you're immediately counting and then just releasing the `Results` object